include/extractor/edge_based_graph_factory.hpp

```diff
@@ -152,14 +157,16 @@ class EdgeBasedGraphFactory
 
     unsigned RenumberEdges();
 
-    std::vector<NBGToEBG> GenerateEdgeExpandedNodes();
+    std::vector<NBGToEBG> GenerateEdgeExpandedNodes(const WayRestrictionMap &way_restriction_map);
```

Comment:
Will document here. You are correct. EdgeExpandedNodes has to generate all our duplicated nodes as well. Therefor it requires access to the way-restriction-map.